pkg/volume/nfs/nfs.go

```diff
@@ -121,7 +122,10 @@ func (plugin *nfsPlugin) newMounterInternal(spec *volume.Spec, pod *v1.Pod, moun
 	if err != nil {
 		return nil, err
 	}
-
+	// wrap ipv6 into `[ ]`
+	if net.ParseIP(source.Server).To16() != nil {
+		source.Server = fmt.Sprintf("[%s]", source.Server)
+	}
```

Comment:
is ok to mutate the field?
should not be better to use something like 
```
        server := source.Server
	if net.ParseIP(source.Server).To16() != nil {
		server = fmt.Sprintf("[%s]", source.Server)
	}
```
and then assign server to the nfsMounter.server?

Reply:
I agree with you that it's better not to mutate, specially that `source.Server` is a `pointer`. 
However in this case, `source.Server` is returned from a private/helper, and the side-effects are bound to the function itself. 

But I agree with you its more safe, a fix on the way, thanks a lot for your review 🙏🏽